File: Producto/backend/src/main/java/backend/com/produccion/application/dto/RecepcionOSDTO.java
package backend.com.produccion.application.dto;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class RecepcionOSDTO {

    private Long idRecepcion;
    private Long osId;
    private LocalDate fechaRecepcion;
    private Integer cantidadRecibida;
    private Integer cantidadConforme;
    private Integer cantidadDefectuosa;
    private String responsable;
    private String observaciones;
}

File: Producto/frontend/src/remote/DTO/RecepcionOSDTO.js
/**
 * DTO para recepción de una OS (recibe el trabajo del taller).
 * Espejo de RecepcionOSDTO.java del backend.
 */
export class RecepcionOSDTO {
    constructor(data = {}) {
        this.idRecepcion = data.idRecepcion ?? null;
        this.osId = data.osId ?? null;
        this.fechaRecepcion = data.fechaRecepcion ?? null;
        this.cantidadRecibida = data.cantidadRecibida ?? 0;
        this.cantidadConforme = data.cantidadConforme ?? 0;
        this.cantidadDefectuosa = data.cantidadDefectuosa ?? 0;
        this.responsable = data.responsable ?? '';
        this.observaciones = data.observaciones ?? '';
    }
}
